### ISSUE TITLE
handleLaneDragEnd is now passed the index of the dragged lane, instead of the lane id, making 2.0.1 a BREAKING change

### ISSUE THREAD

**Issue by G-Rath** [rcdexta/react-trello#119]
The release of `2.0.1` includes a breaking change: `handleLaneDragEnd` is now called with the index of the lane that was 'removed' as the first parameter, instead of the lanes id.

https://github.com/rcdexta/react-trello/blob/65d8dae2e6189c52701b755bd4e41bcb850b4437/src/components/BoardContainer.js#L51-L54

This block of code use to be:

`handleLaneDragEnd(payload.id, addedIndex)`

```
<ReactTrello
    draggable
    data={{
        lanes: [
            { id: 'my-first-lane' },
            { id: 'my-second-lane' },
            { id: 'my-third-lane' }
        ]
    }}

    handleLaneDragEnd={console.log}
/>
```
Resulting console output after dragging a lane:

![ss 2018-10-16 at 08 14 59](https://user-images.githubusercontent.com/3151613/46972700-9cd00200-d11b-11e8-8bb6-024ee12dc7c5.png)

Expected console output should be `my-first-lane 2 [Object object]`

**Comment by G-Rath** [comment]
Weirdly, `handleDragEnd` is unaffected, passing the `id`s I specified for the `sourceLaneId` & `targetLaneId` parameters.

**Comment by CodyFitzpatrick** [comment]
@G-Rath I wonder if card IDs being overridden upon creation (see #117) is due to the same issue?

**Comment by G-Rath** [comment]
@CodyFitzpatrick The issue is relating to `handleLaneDragEnd`: It's now `handleLaneDragEnd(removedIndex, addedIndex, payload)` instead of `handleLaneDragEnd(payload.id, addedIndex)`.

Give me a few minutes to refactor this issue, then we'll talk 😉

**Comment by CodyFitzpatrick** [comment]
Sounds good! Looks like new card IDs are indeed being overwritten in `react-trello/src/components/Lane.js` with the same ID generator function. Here's a snippet of the culprit:

**Comment by CodyFitzpatrick** [comment]
@G-Rath Sounds good! Looks like new card IDs are indeed also being overwritten in react-trello/src/components/Lane.js with the same ID generator function, `uuidv1()`. Here's a snippet of the culprit:
```
addNewCard = params => {
    const laneId = this.props.id
    const id = uuidv1()
    this.hideEditableCard()
    let card = {...params, id}
    this.props.actions.addCard({laneId, card})
    this.props.onCardAdd(card, laneId)
}
```

**Comment by G-Rath** [comment]
As I showed with the original snippet, this'll happen when adding new lanes too:

https://github.com/rcdexta/react-trello/blob/65d8dae2e6189c52701b755bd4e41bcb850b4437/src/helpers/LaneHelper.js#L44-L47

I'll create a new issue for lane ids. I think this library could gain a lot from a bit of clean up in regards to this sort of thing, but I'm just waiting for the authors feedback before spamming the repo with PRs :joy:

Depending on his response, I may start my own fork, as this package is a core dependency of a work project 😕

**Comment by G-Rath** [comment]
This is solved as of v2.0.3
